### PR TITLE
[DO NOT MERGE] Add user event to Laravel

### PR DESCRIPTION
**Pull request by estringana**
### Description

Set user on authenticated request on Laravel framework

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

APPSEC-14732

### CODE REVIEW

File: tests/Integrations/CLI/Laravel/V9_X/CommonScenariosTest.php [hunk 1]
```diff
@@ -6,10 +6,145 @@
 use DDTrace\Tests\Common\SpanAssertion;
 use DDTrace\Tests\Common\CLITestCase;
 
-class CommonScenariosTest extends \DDTrace\Tests\Integrations\CLI\Laravel\V8_X\CommonScenariosTest
+class CommonScenariosTest extends CLITestCase
```
Comment (PROFeNoM) [inline tests/Integrations/CLI/Laravel/V9_X/CommonScenariosTest.php:9]: Why does this have to be changed?

File: tests/Frameworks/Laravel/Version_10_x/config/database.php [hunk 2]
```diff
@@ -46,11 +46,11 @@
...
             'url' => env('DATABASE_URL'),
-            'host' => env('DB_HOST', '127.0.0.1'),
+            'host' => env('DB_HOST', 'mysql_integration'),
             'port' => env('DB_PORT', '3306'),
-            'database' => env('DB_DATABASE', 'forge'),
-            'username' => env('DB_USERNAME', 'forge'),
-            'password' => env('DB_PASSWORD', ''),
+            'database' => env('DB_DATABASE', 'test'),
+            'username' => env('DB_USERNAME', 'test'),
+            'password' => env('DB_PASSWORD', 'test'),
```
Comment (PROFeNoM) [inline tests/Frameworks/Laravel/Version_10_x/config/database.php:53]: I believe these should be set in the `.env`, but I ultimately don't care :)

File: tests/Frameworks/Laravel/Version_10_x/app/Providers/AppServiceProvider.php [hunk 3]
```diff
@@ -19,6 +20,6 @@ public function register(): void
      */
     public function boot(): void
     {
-        //
+        Schema::defaultStringLength(191);
```
Comment (PROFeNoM) [inline tests/Frameworks/Laravel/Version_10_x/app/Providers/AppServiceProvider.php:23]: Just for my understanding and future versions of the framework: What does this have to be set?

File: tests/Frameworks/Laravel/Version_10_x/config/database.php (hunk 2 above)
Comment (estringana) [reply tests/Frameworks/Laravel/Version_10_x/config/database.php:53]: I don't disagree with having this on the `.env`. However I did it this way to keep consistency with the previous versions. Happy to change it if needed

File: tests/Frameworks/Laravel/Version_10_x/app/Providers/AppServiceProvider.php (hunk 3 above)
Comment (estringana) [reply tests/Frameworks/Laravel/Version_10_x/app/Providers/AppServiceProvider.php:23]: It's not common in Laravel having to set this odd thing :( but this is the reason https://laravel.com/docs/master/migrations#index-lengths-mysql-mariadb

File: tests/Frameworks/Laravel/Version_10_x/config/database.php (hunk 2 above)
Comment (PROFeNoM) [reply tests/Frameworks/Laravel/Version_10_x/config/database.php:53]: No that's alright :) It's working. As long as we have some "history of why" in this PR, that's ok for me

**Review by PROFeNoM — APPROVED** [review]
Quite some work there, that's nice :smiley:

**Comment by Anilm3** [comment]
Currently debating the potential PII leaks this could cause and how to address them so let's not merge it for now.

**Pull request closed without merging**